src/IB/IBFESurfaceMethod.cpp

```diff
@@ -672,19 +1037,274 @@ IBFESurfaceMethod::interpolateVelocity(const int u_data_idx,
+                                dx[0] * w[0][0] * w[1][0] *
+                                (LL[0] * DU_j_qp[d][s * NDIM] + LL[1] * DU_j_qp[d][1 + s * NDIM]);
+                            Ujump[ic_upper[0]][ic_lower[1]][d] =
+                                dx[0] * w[0][1] * w[1][0] *
+                                (UL[0] * DU_j_qp[d][s * NDIM] + UL[1] * DU_j_qp[d][1 + s * NDIM]);
+                            Ujump[ic_upper[0]][ic_upper[1]][d] =
+                                dx[0] * w[0][1] * w[1][1] *
+                                (UU[0] * DU_j_qp[d][s * NDIM] + UU[1] * DU_j_qp[d][1 + s * NDIM]);
+                            Ujump[ic_lower[0]][ic_upper[1]][d] =
+                                dx[0] * w[0][0] * w[1][1] *
+                                (LU[0] * DU_j_qp[d][s * NDIM] + LU[1] * DU_j_qp[d][1 + s * NDIM]);
+                        }
+#endif
+#if (NDIM == 3)
+                        for (int d = 0; d < NDIM; ++d)
+                        {
+                            LLL[d] = (n_qp[s * NDIM] * wr[0][1] + n_qp[s * NDIM + 1] * wr[1][1] +
+                                      n_qp[s * NDIM + 2] * wr[2][1]) *
+                                     n_qp[s * NDIM + d];
+                            LLU[d] = (n_qp[s * NDIM] * wr[0][1] + n_qp[s * NDIM + 1] * wr[1][1] +
+                                      n_qp[s * NDIM + 2] * wr[2][0]) *
+                                     n_qp[s * NDIM + d];
+                            LUL[d] = (n_qp[s * NDIM] * wr[0][1] + n_qp[s * NDIM + 1] * wr[1][0] +
+                                      n_qp[s * NDIM + 2] * wr[2][1]) *
+                                     n_qp[s * NDIM + d];
+                            ULL[d] = (n_qp[s * NDIM] * wr[0][0] + n_qp[s * NDIM + 1] * wr[1][1] +
+                                      n_qp[s * NDIM + 2] * wr[2][1]) *
+                                     n_qp[s * NDIM + d];
+                            UUL[d] = (n_qp[s * NDIM] * wr[0][0] + n_qp[s * NDIM + 1] * wr[1][0] +
+                                      n_qp[s * NDIM + 2] * wr[2][1]) *
+                                     n_qp[s * NDIM + d];
+                            ULU[d] = (n_qp[s * NDIM] * wr[0][0] + n_qp[s * NDIM + 1] * wr[1][1] +
+                                      n_qp[s * NDIM + 2] * wr[2][0]) *
+                                     n_qp[s * NDIM + d];
+                            UUU[d] = (n_qp[s * NDIM] * wr[0][0] + n_qp[s * NDIM + 1] * wr[1][0] +
+                                      n_qp[s * NDIM + 2] * wr[2][0]) *
+                                     n_qp[s * NDIM + d];
+                            LUU[d] = (n_qp[s * NDIM] * wr[0][1] + n_qp[s * NDIM + 1] * wr[1][0] +
+                                      n_qp[s * NDIM + 2] * wr[2][0]) *
+                                     n_qp[s * NDIM + d];
+                        }
+                        for (int d = 0; d < NDIM; ++d)
+                        {
+                            Ujump[ic_lower[0]][ic_lower[1]][ic_lower[2]][d] =
```

Comment:
If possible, I would like these very complicated looking statements to be transformed into simpler loops. This kind of code can be tough to maintain, and I don't think we need the performance that can come with manually unrolling as done here.

Reply:
Yes. If the loop bounds are given to the compiler then it will unroll when appropriate.

Reply:
This is not about efficiency --- these should not be dominate computational costs, because they are happening on an `NDIM-1` surface in an `NDIM` computation. What I want to try to accomplish is to enhance the readability of the implementation.

Towards that goal, @AminKolah, it would be helpful if you are doing things in "normal" and "tangential" directions (or looking at "normal" and "tangential" indices, etc.) that you use variable names that include that information --- e.g. i_norm and j_tan.

For indices, it would be fairly natural to do something like:
```
    int i_norm = ... ;
    vector<int> j_tan = (NDIM == 2 ? { (i_norm + 1) % NDIM } : { (i_norm + 1) % NDIM , (i_norm + 2) % NDIM});
```

Reply:
Okay I just started working on turning these into simple/more compact loops in the interpolateVelocity function. Would you also recommend using things like inner_product instead of component by component multiplication?

Reply:
Definitely!

If you use either libMesh or Eigen vectors, you should be able to do an inner product using `*`. My goal in requesting these changes is to get the expressions to look more like "math". :-)